src/lstm_layer.cpp

```diff
@@ -1380,6 +1534,19 @@ void LSTM::forward(BaseHiddenStates &input_states,
         lstm_to_prev_states(this->lstm_states.var_c,
                             this->lstm_states.var_c_prior.size(),
                             this->lstm_states.var_c_prior);
+
+        // Save priors for smoothing
+        this->lstm_states.mu_h_priors.push_back(this->lstm_states.mu_h_prior);
```

Comment:
Do we want to use smoother permanently for LSTM layer? If not, you need to create a if-statement to turn on-and-off for that option.

Reply:
I am thinking of creating a SmootherLSTM or `SLSTM` layer that is different than LSTM because it will avoid introducing bugs. If there are a lot of overlapped code between SMLSTM and LSTM (>50%), you should create the common functions stored in  lstm_layer.cpp and called them from SLSTM layer

Reply:
ok. I will do that